bindings/python/release.sh

```diff
@@ -7,26 +7,21 @@ set -o errexit
 PYTHON=${PYTHON_BINARY:-"python"}
 $PYTHON --version
 
-# Version of libbson to use
-# Keep in sync with pymongoarrow.version._MIN_LIBBSON_VERSION
-LIBBSON_REVISION=${LIBBSON_VERSION:-"1.17.5"}
-echo "Using libbson $LIBBSON_REVISION"
-
 # Compute shared library filename
 if [ "Darwin" = "$(uname -s)" ]
 then
   SO_EXT='dylib'
 elif [ "Darwin" = "$(uname -s)" ]
 then
   SO_EXT='so'
-  export MACOSX_DEPLOYMENT_TARGET="10.15"
+  export MACOSX_DEPLOYMENT_TARGET="10.9"
```

Comment:
This gets rid of the `object file . . . was built for newer OSX version (10.xx) than being linked (10.yy)` warnings. I haven't been able to produce an error, but I think our extensions will fail without this because we use C++ (it doesn't affect C so extensions that are C-only are not affected).